collector/src/main/java/io/prometheus/jmx/JmxScraper.java

```diff
@@ -89,7 +90,16 @@ public void doScrape() throws Exception {
         MBeanServerConnection beanConn;
         JMXConnector jmxc = null;
         if (jmxUrl.isEmpty()) {
-          beanConn = ManagementFactory.getPlatformMBeanServer();
+          try {
```

Comment:
Throwing an exception is the correct behaviour, as that'll make the scrape fail.